src/test/java/cat/udl/eps/softarch/geolearning/steps/CreateMatchResultStepDefs.java

```diff
@@ -0,0 +1,110 @@
+package cat.udl.eps.softarch.geolearning.steps;
+
+import cat.udl.eps.softarch.geolearning.domain.Match;
+import cat.udl.eps.softarch.geolearning.domain.MatchResult;
+import cat.udl.eps.softarch.geolearning.domain.Player;
+import cat.udl.eps.softarch.geolearning.repository.MatchRepository;
+import cat.udl.eps.softarch.geolearning.repository.MatchResultRepository;
+import cat.udl.eps.softarch.geolearning.repository.PlayerRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class CreateMatchResultStepDefs {
+
+    private Player player1;
+    private Match match;
+    private MatchResult matchResult;
+    //private MatchResult matchResultSaved;
+
+    @Autowired
+    private PlayerRepository playerRepository;
+
+    @Autowired
+    private MatchRepository matchRepository;
+
+    @Autowired
+    private MatchResultRepository matchResultRepository;
+
+    @Given("There is a player {string} with password {string} and {string}")
+    public void thereIsAPlayerWithPasswordAnd(String username, String password, String email) {
+        if(!playerRepository.existsById(username))
+        {
+            player1 = new Player();
+            player1.setUsername(username);
+            player1.setPassword(password);
+            player1.setEmail(email);
+            player1.encodePassword();
+            playerRepository.save(player1);
+        }
+    }
+
+    @And("There is a match with id {int}")
+    public void thereIsAMatchWithId(Integer matchId) {
+        if(!matchRepository.existsById(matchId))
+        {
+            match = new Match();
+            match.setId(matchId);
+            matchRepository.save(match);
+        }
+    }
+
+    @Given("There is no registered matchResult for this match attached to the player")
+    public void thereIsNoRegisteredMatchResultForThisMatchAttachedToThePlayer() {
+        Assert.assertNull(matchResultRepository.findByMatchAndPlayer(match, player1));
+    }
+
+    @Given("We played the match previously")
+    public void wePlayedTheMatchPreviously() {
+        matchResult = new MatchResult();
+        matchResult.setResult(1);
+        matchResult.setTime(1);
+        matchResult.setMatch(match);
+        matchResult.setPlayer(player1);
+        matchResultRepository.save(matchResult);
+    }
+
+    @When("The match is finished with a matchResult with result {int} and time {int}")
+    public void theMatchIsFinishedWithAMatchResultWithResultAndTime(int result, int time) {
```

Comment:
This is not testing the REST API. From the client side, we will not be able to call Java repositories. Instead, we will create the MatchResult doing a POST to /matchResults and with this test, we don't know if it is working!